src/main/java/org/assertj/vavr/internal/Maps.java

```diff
@@ -35,16 +39,42 @@
 
     private Failures failures = Failures.instance();
 
+    private Conditions conditions = Conditions.instance();
+
     private Maps() {}
 
     public static Maps instance() {
         return INSTANCE;
     }
 
+    /**
+     * Verifies that the given {@code Map} contains the value for given {@code key} that satisfy given {@code valueCondition}.
+     *
+     * @param <K>            key type
+     * @param <V>            value type
+     * @param info           contains information about the assertion.
+     * @param actual         the given {@code Map}.
+     * @param key            the given key to check.
+     * @param valueCondition the given condition for check value.
+     * @throws NullPointerException if the given values is {@code null}.
+     * @throws AssertionError       if the actual map is {@code null}.
+     * @throws AssertionError       if the actual map not contains the given {@code key}.
+     * @throws AssertionError       if the actual map contains the given key, but value not match the given {@code valueCondition}.
+     */
+    @SuppressWarnings("unchecked")
+    public <K, V> void assertHasEntrySatisfying(AssertionInfo info, Map<K, V> actual, K key,
+                                                Condition<? super V> valueCondition) {
+        assertContainsKeys(info, actual, key);
+        conditions.assertIsNotNull(valueCondition);
```

Comment:
Maybe it's better idea to make this assertion first? It's fast and it fails you don't need to verify the keys.

Reply:
Good idea